test/utils/utils_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"testing"
+)
+
+func TestCheckField(t *testing.T) {
+	testCases := []struct {
+		desc        string
+		input       string
+		regexString string
+		expectVal   string
+		expectFound bool
+	}{
+		{
+			desc:        "Input contains a match",
+			input:       "--currentNodeCount: 3\n--",
+			regexString: `currentNodeCount: (.*)\n`,
```

Comment:
This test is a bit weird. if you are testing the use of this **specific** regex, you need to make this regex a const in the code so there is not possibility the use in this test diverges with the one in the code.

There is **no** connection between the code and this test.

regexString also is the same value in each test.